starboard/nplb/player_write_sample_test.cc

```diff
@@ -111,14 +112,204 @@ TEST_P(SbPlayerWriteSampleTest, WriteMultipleBatches) {
+    samples.AddAudioSamples(
+        0, player_fixture.ConvertDurationToAudioBufferCount(kSbTimeSecond));
+    samples.AddAudioEOS();
+  }
+  if (player_fixture.HasVideo()) {
+    samples.AddVideoSamples(
+        0, player_fixture.ConvertDurationToVideoBufferCount(kSbTimeSecond));
+    samples.AddVideoEOS();
+  }
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(samples));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerEndOfStream());
+}
+
+TEST_P(SbPlayerWriteSampleTest, PartialAudio) {
+  if (!IsPartialAudioSupported()) {
+    // TODO: Use GTEST_SKIP when we have a newer version of gtest.
+    SB_LOG(INFO)
+        << "The platform doesn't support partial audio. Skip the tests.";
+    return;
+  }
+
+  SbPlayerTestFixture player_fixture(GetParam(),
+                                     &fake_graphics_context_provider_);
+  if (HasFatalFailure()) {
+    return;
+  }
+  if (!player_fixture.HasAudio()) {
+    // TODO: Use GTEST_SKIP when we have a newer version of gtest.
+    SB_LOG(INFO) << "Skip PartialAudio test for audioless content.";
+    return;
+  }
+
+  const SbTime kDurationToPlay = kSbTimeSecond;
+  const float kSegmentSize = 0.1f;
+
+  GroupedSamples samples;
+  if (player_fixture.HasVideo()) {
+    samples.AddVideoSamples(
+        0, player_fixture.ConvertDurationToVideoBufferCount(kDurationToPlay));
+    samples.AddVideoEOS();
+  }
+
+  int total_buffers_to_write =
+      player_fixture.ConvertDurationToAudioBufferCount(kDurationToPlay);
+  for (int i = 0; i < total_buffers_to_write; i++) {
+    SbTime current_timestamp = player_fixture.GetAudioSampleTimestamp(i);
+    SbTime next_timestamp = player_fixture.GetAudioSampleTimestamp(i + 1);
+    SbTime buffer_duration = next_timestamp - current_timestamp;
+    SbTime segment_duration = buffer_duration * kSegmentSize;
+    SbTime written_duration = 0;
+    while (written_duration < buffer_duration) {
+      samples.AddAudioSamples(
+          i, 1, written_duration, written_duration,
+          std::max<int64_t>(
+              0, buffer_duration - written_duration - segment_duration));
+      written_duration += segment_duration;
+    }
+  }
+  samples.AddAudioEOS();
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(samples));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerPresenting());
+
+  SbTime start_system_time = SbTimeGetMonotonicNow();
+  SbTime start_media_time = player_fixture.GetCurrentMediaTime();
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerEndOfStream());
+
+  SbTime end_system_time = SbTimeGetMonotonicNow();
+  SbTime end_media_time = player_fixture.GetCurrentMediaTime();
+
+  const SbTime kDurationDifferenceAllowance = 500 * kSbTimeMillisecond;
+  EXPECT_NEAR(end_media_time, kDurationToPlay, kDurationDifferenceAllowance);
+  EXPECT_NEAR(end_system_time - start_system_time + start_media_time,
+              kDurationToPlay, kDurationDifferenceAllowance);
+
+  SB_DLOG(INFO) << "The expected media time should be " << kDurationToPlay
```

Comment:
Yes, that is to address Xiaoming's comment https://github.com/youtube/cobalt/pull/545#discussion_r1231696109. EXPECT_NEAR is used, so it would print the actual media time and running time for every test run. The result is same as adding log before ASSERT_NEAR.

Reply:
Ack